contracts/IronVest.sol

```diff
@@ -1,761 +1,732 @@
+        UserCliffInfo storage info = userCliffInfo[_poolId][_msgSender()];
+        require(
+            cliffPoolInfo[_poolId].cliffPeriodEndTime < block.timestamp,
+            "IIronVest : Cliff Period Is Not Over Yet"
+        );
+
+        uint256 transferAble = cliffClaimable(_poolId, _msgSender());
+        require(transferAble > 0, "IIronVest : Invalid TransferAble");
+        IERC20Upgradeable(cliffPoolInfo[_poolId].tokenAddress).safeTransfer(
+            _msgSender(),
+            transferAble
+        );
+        uint256 claimed = transferAble + info.claimedAmnt;
+        uint256 remainingTobeClaimable = info.cliffAlloc - claimed;
+        info.claimedAmnt = claimed;
+        info.remainingToBeClaimableCliff = remainingTobeClaimable;
+        info.cliffLastWithdrawal = block.timestamp;
+
+        emit CliffClaim(
+            _poolId,
+            transferAble,
+            _msgSender(),
+            remainingTobeClaimable
+        );
+    }
+
+    /// @dev User must have allocation in the pool.
+    /// @notice This is for claiming cliff vesting.
+    /// @notice should be called if need to claim non cliff amount.
+    /// @param _poolId : Pool Id from which pool user want to withdraw.
+    /// @notice Secured by nonReentrant.
+    function claimNonCliff(uint256 _poolId) external nonReentrant {
+        UserNonCliffInfo storage info = userNonCliffInfo[_poolId][_msgSender()];
+        require(
+            cliffPoolInfo[_poolId].cliffPeriodEndTime < block.timestamp,
+            "IIronVest : Cliff Period Is Not Over Yet"
+        );
+
+        uint256 transferAble = nonCliffClaimable(_poolId, _msgSender());
+        uint256 claimed = transferAble + info.claimedAmnt;
+        require(transferAble > 0, "IIronVest : Invalid TransferAble");
+        IERC20Upgradeable(cliffPoolInfo[_poolId].tokenAddress).safeTransfer(
+            _msgSender(),
+            transferAble
+        );
+        uint256 remainingTobeClaimable = info.nonCliffAlloc - claimed;
+        info.claimedAmnt = claimed;
+        info.remainingToBeClaimableNonCliff = remainingTobeClaimable;
+        info.nonCliffLastWithdrawal = block.timestamp;
+        emit NonCliffClaim(
+            _poolId,
+            transferAble,
+            _msgSender(),
+            remainingTobeClaimable
+        );
+    }
+
+    /// @dev this function use to withdraw tokens that send to the contract mistakenly
+    /// @param _token : Token address that is required to withdraw from contract.
+    /// @param _amount : How much tokens need to withdraw.
+    function emergencyWithdraw(IERC20Upgradeable _token, uint256 _amount)
+        external
+        onlyAdmin
+    {
+        IERC20Upgradeable(_token).safeTransfer(_msgSender(), _amount);
+    }
+
+    /// @dev Functions is called by a default admin.
+    /// @param _signer : An address whom admin want to be a signer.
+    function setSigner(address _signer) external onlyAdmin {
```

Comment:
@SibghatUllah1997 this one should be onlyOwner. I don't see onlyOwner modifier anywhere actually. The owner by default will be the address that is deploying the contract.